### ISSUE TITLE
sympy.Matrix.inv has numerical problems on invertible matrix (compared with numpy)

### ISSUE THREAD

**Issue by boeddeker** [sympy/sympy#11549]
The Matrix is invertible, but the inverse of sympy.Matrix.inv is wrong, if the top left value is 6e-17

Code: 

```
import numpy as np, sympy
print(sympy.__version__)

A_np = np.array([
[6e-17, 1.0, 4.0],
[ -1.0,   0, 8.0],
[    0,   0, 1.0]])

print('np: ', np.linalg.inv(A_np))
print('inv: ', repr(sympy.Matrix(A_np).inv()))
print('pinv: ', repr(sympy.Matrix(A_np).pinv()))

A_np = np.array([
[    0, 1.0, 4.0],
[ -1.0,   0, 8.0],
[    0,   0, 1.0]])

print('np: ', np.linalg.inv(A_np))
print('inv: ', repr(sympy.Matrix(A_np).inv()))
print('pinv: ', repr(sympy.Matrix(A_np).pinv()))
```

￼

Output: 
￼

> 1.0
> np:  [[ -0.00000000e+00  -1.00000000e+00   8.00000000e+00]
>  [  1.00000000e+00   6.00000000e-17  -4.00000000e+00]
>  [  0.00000000e+00   0.00000000e+00   1.00000000e+00]]
> inv:  Matrix([
> [  0,    -1.0, **_16.0_**],
> [1.0, 6.0e-17, -4.0],
> [  0,       0,  1.0]])
> pinv:  Matrix([
> [  0, -1.0,  8.00000000000006],
> [1.0,    0, -4.00000000000003],
> [  0,    0,  1.00000000000001]])
> np:  [[-0. -1.  8.]
>  [ 1.  0. -4.]
>  [ 0.  0.  1.]]
> inv:  Matrix([
> [  0, -1.0,  8.0],
> [1.0,    0, -4.0],
> [  0,    0,  1.0]])
> pinv:  Matrix([
> [  0, -1.0,  8.0],
> [1.0,    0, -4.0],
> [  0,    0,  1.0]])

**Comment by isuruf** [comment]
For numerical problems, you should use numpy. If you want to get the same answer, you can do `sympy.Matrix(A_np).n(30).inv().n(16)` which uses higher precision floats so that the numerical is reduced. In this case you can also use, sympy.Matrix(A_np).inverse_ADJ()`

**Comment by boeddeker** [comment]
ok, that solves my problem, but the thing what I wanted to report is, that a value of 6.0e-17 for a Matrix which is good invertable (determinante is one) results in an error of 8 (should be 8, but is 16).

The value of  6.0e-17 appears in my calculations (numerical error, the analytical value is 0) and if it is a bit greater ( + 2.0e-17 = 8.0e-17 ) the solution is correct.

**Comment by siefkenj** [comment]
I suspect if sympy used partial pivoting during Gaussian elimination, it would reduce these numerical errors.

**Comment by cbm755** [comment]
I agree: I've tested this and it seems fixed thanks to great work of @siefkenj (#11554) and others.
